test/functional/runner/index.js

```diff
@@ -503,6 +516,14 @@ function testOperation(operation, obj, context, options) {
     opPromise = Promise.try(() => obj[operationName].apply(obj, args));
   }
 
+  if (operation.error) {
+    opPromise = opPromise
+      .then(() => {
+        throw new Error('expected an error!');
+      })
+      .catch(() => {});
```

Comment:
It seems like what you want is:

```js
opPromise = opPromise.then(
  () => {
    throw new Error('expected an error!');
  },
  () => {}
);
```

putting the `catch` after the `then` means this will always resolve and never reject.

Reply:
ah, yeah I see your point. fixed

Reply:
Fixed this for a similar case below as well